flareon/src/test.rs

```diff
@@ -179,44 +181,118 @@ impl TestRequestBuilder {
 }
 
 #[derive(Debug)]
-pub struct TestDatabaseBuilder {
+pub struct TestDatabase {
+    database: Arc<Database>,
+    kind: TestDatabaseKind,
     migrations: Vec<MigrationWrapper>,
 }
 
-impl Default for TestDatabaseBuilder {
-    fn default() -> Self {
-        Self::new()
-    }
-}
-
-impl TestDatabaseBuilder {
-    #[must_use]
-    pub fn new() -> Self {
+impl TestDatabase {
+    fn new(database: Database, kind: TestDatabaseKind) -> TestDatabase {
         Self {
+            database: Arc::new(database),
+            kind,
             migrations: Vec::new(),
         }
     }
 
-    #[must_use]
+    /// Create a new in-memory SQLite database for testing.
+    pub async fn new_sqlite() -> Result<Self> {
+        let database = Database::new("sqlite::memory:").await?;
+        Ok(Self::new(database, TestDatabaseKind::Sqlite))
+    }
+
+    /// Create a new Postgres database for testing and connects to it.
+    ///
+    /// The database URL is read from the `POSTGRES_URL` environment variable.
+    /// Note that it shouldn't include the database name — the function will
+    /// create a new database for the test by connecting to the `postgres`
+    /// database.
+    ///
+    /// The database is created with the name `test_flareon__{test_name}`.
+    /// Make sure that `test_name` is unique for each test so that the databases
+    /// don't conflict with each other.
+    ///
+    /// The database is dropped when `self.cleanup()` is called. Note that this
+    /// means that the database will not be dropped if the test panics.
```

Comment:
Could we try catching a panic to be able to always cleanup the database?

Reply:
I was thinking about it and I think there's a value in not cleaning up the database. If a test fails, you can see the state of the DB to see what happened. It will be cleaned up after another run of the test (there is a "DROP DATABASE IF EXISTS" at the beginning of the test) anyway. What do you think?

Reply:
You're right, there's value in being able to see the broken state; however, cleanup phase is also for the graceful disconnection and not leaving dangling connections. Having said that, those are databases spun up for testing purposes, so I assume there's not really an impact here.